### PR TITLE
Add Application Name to PostgreSQL Connection String

### PR DESCRIPTION
**Pull request by stefanbenten**
This PR adds a proper Application Name to the PostgreSQL Connection String, if none is provided

### CODE REVIEW

**Comment by stefanbenten** [comment]
@egonelbre Shoot, navigated into the wrong file...
Relocated.

File: satellite/satellitedb/database.go [hunk 1]
```diff
@@ -41,7 +43,9 @@ func New(log *zap.Logger, databaseURL string) (satellite.DB, error) {
 	if err != nil {
 		return nil, err
 	}
-
+	if !strings.Contains(source, "ApplicationName") {
+		source = fmt.Sprintf("%s?%s", source, "ApplicationName=Satellite")
```
Comment (egonelbre) [inline satellite/satellitedb/database.go]: the `source` may already contain query parameters and the driver can here be sqlite as well

File: internal/dbutil/pgutil/db.go [hunk 2]
```diff
@@ -98,3 +100,17 @@ func QuerySnapshot(db dbschema.Queryer) (*dbschema.Snapshot, error) {
 		Data:    data,
 	}, err
 }
+
+//CheckApplicationName ensures that the Connection String contains an application name
+func CheckApplicationName(s string) (r string) {
+	if !strings.Contains(s, "ApplicationName") {
```
Comment (aleitner) [inline internal/dbutil/pgutil/db.go]: shouldn't you be checking for `application_name` instead of `ApplicationName`?

Comment (stefanbenten) [reply internal/dbutil/pgutil/db.go]: Done, good catch!

File: internal/dbutil/pgutil/db.go [hunk 3]
```diff
@@ -98,3 +100,17 @@ func QuerySnapshot(db dbschema.Queryer) (*dbschema.Snapshot, error) {
...
+func CheckApplicationName(s string) (r string) {
+	if !strings.Contains(s, "application_name") {
+		if !strings.Contains(s, "?") {
+			r = fmt.Sprintf("%s?%s", s, "application_name=Satellite")
+		} else {
+			r = fmt.Sprintf("%s%s%s", s, "&", "application_name=Satellite")
+		}
+		return
+	}
+	//return source as is if ApplicationName is set
```
Comment (aleitner) [inline internal/dbutil/pgutil/db.go]: Doesn't really need to be fixed but you could change this comment too

File: internal/dbutil/pgutil/db.go [hunk 4]
```diff
@@ -98,3 +100,17 @@ func QuerySnapshot(db dbschema.Queryer) (*dbschema.Snapshot, error) {
...
 	}, err
 }
+
+//CheckApplicationName ensures that the Connection String contains an application name
+func CheckApplicationName(s string) (r string) {
+	if !strings.Contains(s, "application_name") {
+		if !strings.Contains(s, "?") {
+			r = fmt.Sprintf("%s?%s", s, "application_name=Satellite")
+		} else {
+			r = fmt.Sprintf("%s%s%s", s, "&", "application_name=Satellite")
```
Comment (wthorp) [inline internal/dbutil/pgutil/db.go]: nit:
you could really just use string concatenation here and above.
you could also directly return here and above.

Comment (stefanbenten) [reply internal/dbutil/pgutil/db.go]: Sure, let me change that real quick!

**Pull request merged**